ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -495,4 +511,37 @@ private void displayWelcomeMessage() {
 
     writer.flush();
   }
+
+  static void maybeCreateProcessingLogStream(
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine,
+      final CommandQueue commandQueue) {
+    if (!config.getString(ProcessingLogConfig.STREAM_AUTO_CREATE).equals(
+        ProcessingLogConfig.AUTO_CREATE_ON)) {
+      return;
+    }
+    final String name = config.getString(ProcessingLogConfig.STREAM_NAME);
+    final String topicName = config.getString(ProcessingLogConfig.TOPIC_NAME);
+    final String statementText = ProcessingLogServerUtils.processingLogStreamCreateStatement(
+        name,
+        topicName);
+    if (!commandQueue.isEmpty()) {
+      return;
+    }
+    final List<PreparedStatement<?>> statement;
+    try {
+      statement = ksqlEngine.parseStatements(statementText);
+      ksqlEngine.tryExecute(statement, ksqlConfig, Collections.emptyMap());
+    } catch (final KsqlException e) {
+      log.warn("Failed to create processing log stream", e);
+      return;
+    }
+    commandQueue.enqueueCommand(
```

Comment:
This is queuing a command to the command topic on each server start, for each server in the cluster, right?

Is this intentional? Why not just execute this against the engine directly as we do for the command topic?

Reply:
I want it on the command topic because the schema written by the logger is going to evolve over time, and there's a risk that it could do so in a way that's incompatible with downstream queries that get written to the command topic, which would then corrupt the meta-store during recovery. For example:

Suppose the current ddl is:
```
CREATE STREAM PROCESSING_LOG (A STRUCT<...>, B STRUCT<...>) ...
```

And the user does:
```
CREATE STREAM ENRICHED AS SELECT *, SOMEFUNC(A) AS C FROM PROCESSING_LOG;
```

And then later on we extend the processing log schema to have a struct C, the user's statement would fail validation when we reboot.

The other option I considered here was only allowing transient queries on this stream, but that seems unnecessarily restrictive. On the other hand you do always get the latest/greatest schema.

> This is queuing a command to the command topic on each server start, for each server in the cluster, right?

It only enqueues the command if the command topic is empty, so you will get anywhere from 1 to # nodes in cluster instances of the command on the topic. But getting more than 1 is pretty unlikely and is benign.

Reply:
So... you're saying that the schema of the processing log for a running cluster never evolves?  Ouch.  It seems a shame to deny upgraded interactive clusters from taking advantage of the new fields just because we're worried about name clashes on schema changes.

Maybe we could just introduce a reserved prefix for identifiers that we can use, e.g. '_'.  That way we can extend the schema without any worry of conflicts.  We can also avoid writing to the command topic.

Reply:
> So... you're saying that the schema of the processing log for a running cluster never evolves? Ouch. It seems a shame to deny upgraded interactive clusters from taking advantage of the new fields just because we're worried about name clashes on schema changes.

It doesn't evolve automatically. We publish the DDL in the docs, so the user can always drop the automatically created stream and then create one with the new schema.

> Maybe we could just introduce a reserved prefix for identifiers that we can use, e.g. '_'. That way we can extend the schema without any worry of conflicts. We can also avoid writing to the command topic.

I'm not sure here. There's some weird edge cases. Need to think about that a bit more.

Reply:
OK, fair point.